docs/articles/semantic_search_news.md

```diff
@@ -0,0 +1,330 @@
+    text=news.description, model="sentence-transformers/all-mpnet-base-v2"
+)
+headline_space = TextSimilaritySpace(
+    text=news.headline, model="sentence-transformers/all-mpnet-base-v2"
+)
+# release date is encoded using our recency embedding algorithm
+recency_space = RecencySpace(
+    timestamp=news.release_timestamp,
+    period_time_list=[
+        PeriodTime(timedelta(days=4 * YEAR_IN_DAYS), weight=1),
+        PeriodTime(timedelta(days=11 * YEAR_IN_DAYS), weight=2),
+    ],
+    negative_filter=0.0,
+)
+```
+
+```python
+# we create an index of our spaces
+news_index = Index(spaces=[description_space, headline_space, recency_space])
+```
+
+```python
+# simple query will serve us right when we simply want to search the dataset with a search term
+# the term will search in both textual fields
+# and we will have to option to weight certain inputs' importance
+simple_query = (
+    Query(
+        news_index,
+        weights={
+            description_space: Param("description_weight"),
+            headline_space: Param("headline_weight"),
+            recency_space: Param("recency_weight"),
+        },
+    )
+    .find(news)
+    .similar(description_space.text, Param("query_text"))
+    .similar(headline_space.text, Param("query_text"))
+    .limit(Param("limit"))
+)
+
+# news query on the other hand will search in the database with the vector of a news article
+# weighting possibility is still there
+news_query = (
+    Query(
+        news_index,
+        weights={
+            description_space: Param("description_weight"),
+            headline_space: Param("headline_weight"),
+            recency_space: Param("recency_weight"),
+        },
+    )
+    .find(news)
+    .with_vector(news, Param("news_id"))
+    .limit(Param("limit"))
+)
+```
+
+```python
+dataframe_parser = DataFrameParser(
+    schema=news,
+    mapping={news.release_timestamp: "date", news.description: "short_description"},
+)
+```
+
+```python
+source: InMemorySource = InMemorySource(news, parser=dataframe_parser)
+executor: InMemoryExecutor = InMemoryExecutor(
+    sources=[source], indices=[news_index], context_data=EXECUTOR_DATA
+)
+app: InMemoryApp = executor.run()
+```
+
+```python
+source.put([business_news])
+```
+
+## Understanding recency
+
+```python
+recency_plotter = RecencyPlotter(recency_space, context_data=EXECUTOR_DATA)
+recency_plotter.plot_recency_curve()
+```
+
+## Queries
+
+```python
+# quick helper to present the results in a notebook
+def present_result(
+    result_to_present: Result,
+    cols_to_keep: list[str] | None = None,
+) -> pd.DataFrame:
+    if cols_to_keep is None:
+        cols_to_keep = [
+            "description",
+            "headline",
+            "release_date",
+            "id",
+            "similarity_score",
+        ]
+    # parse result to dataframe
+    df: pd.DataFrame = result_to_present.to_pandas()
+    # transform timestamp back to release year. Ts is in milliseconds originally hence the division
```

Comment:
in milliseconds at what point? unix timestamp? or in original dataset?

Reply:
in the original dataset